### DISCUSSION TITLE
.NET/C# POCOs & Power Fx interop?

### DISCUSSION THREAD

**Discussion by mikoskinen** [microsoft/Power-Fx#57] [General]
Does Power Fx currently support (or is there a plan to support) passing .NET objects to Power Fx functions? I see that there is some work related to CustomObjects and there is support for RecordTypes, which I think could be used to represent .NET objects, but couldn't get a clear picture of the situation.

The use case is that we are thinking of using the Power Fx as a user editable "scripting language" inside our .NET app. In this scenario we could have something like List of customers (in .NET side) and we would like to run Power Fx functions like Lookup, Average, Sort etc. against these objects.

**Comment by MikeStall** [comment]
You can indeed pass a .net object into Power Fx. It currently does a deep-copy and gets marshalled as a record. That means no recursive objects yet, and you can't get your original object back out. 
 
We are working on better C#/Power Fx marshalling.
